be/src/exec/pipeline/exchange/local_exchange_source_operator.cpp

```diff
@@ -43,6 +49,28 @@ bool LocalExchangeSourceOperator::has_output() const {
            (_is_finished && _partition_rows_num > 0);
 }
 
+void LocalExchangeSourceOperator::finish_backward(RuntimeState* state) {
+    std::lock_guard<std::mutex> l(_chunk_lock);
+    if (_is_finished) {
+        return;
+    }
+    _is_finished = true;
+    // Compute out the number of rows of the _full_chunk_queue.
+    size_t full_rows_num = 0;
+    while (!_full_chunk_queue.empty()) {
+        auto&& chunk = std::move(_full_chunk_queue.front());
+        _full_chunk_queue.pop();
+        full_rows_num += chunk->num_rows();
+    }
+    // clear _full_chunk_queue
+    { [[maybe_unused]] typeof(_full_chunk_queue) tmp = std::move(_full_chunk_queue); }
```

Comment:
Why use this strange way to clear the queue?

Reply:
queue have no clear function